OpenSim/Common/Component.h

```diff
@@ -480,6 +480,14 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     obtain ground- and body-fixed geometry (with \a fixed=\c true), and then 
     once per frame (with \a fixed=\c false) to generate on-the-fly geometry such
     as rubber band lines, force arrows, labels, or debugging aids.
+
+    Please note, that the state passed to generateDecorations is only guaranteed
+    to be realized to Stage::Position. If your component can visualize quantities
```

Comment:
How is the `Position` guarantee enforced?

Reply:
*Guarantee* is perhaps too strong. Just during `Model::initSystem()` we realize to `Position` so any use immediate after getting the state will be guaranteed. 

This is of course not the case if copy the state or do other things.  There is an expectation that state is at `Position` stage since all frames, bodies and points are being drawn with out explicit realization to `Position`.

From the point of view or our Tools and what they do, IK states are realized to `Position`, AnalyzeTool to `Velocity` and `ForwardTool` uses an Integrator that realizes the current state to `Acceleration`. In the case of a the `ForwardTool` it computes the next state and that state I believe is only realized to `Time`.

Should I soften the language? We can also make the drawing code realize to  `Position` explicitly.

Reply:
I think you could say that there is a precondition that the state is realized to `Position`. No need for explicitly realizing.

Reply:
Done.